File: app/src/main/java/antoninovitale/dropcodechallenge/details/BeerDetailsViewModel.java
package antoninovitale.dropcodechallenge.details;

import android.arch.lifecycle.ViewModel;

/**
 * Created by antoninovitale on 02/09/2017.
 */
class BeerDetailsViewModel extends ViewModel {
    private BeerDetailsPresenter presenter;

    BeerDetailsPresenter getPresenter() {
        return presenter;
    }

    void setPresenter(BeerDetailsPresenter presenter) {
        if (this.presenter == null) {
            this.presenter = presenter;
        }
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        presenter = null;
    }

}

File: app/src/main/java/com/ninovitale/punkapi/app/details/BeerDetailsViewModel.kt
package com.ninovitale.punkapi.app.details

import androidx.lifecycle.ViewModel

/**
 * Created by antoninovitale on 02/09/2017.
 */
class BeerDetailsViewModel : ViewModel() {
    private var presenter: BeerDetailsPresenter? = null
    fun getPresenter(): BeerDetailsPresenter? {
        return presenter
    }

    fun setPresenter(presenter: BeerDetailsPresenter?) {
        if (this.presenter == null) {
            this.presenter = presenter
        }
    }

    override fun onCleared() {
        super.onCleared()
        presenter = null
    }
}